datapackage.go

```diff
@@ -5,21 +5,34 @@ package datapackage
 import (
 	"archive/tar"
 	"compress/gzip"
-	"errors"
 	"io"
 	"os"
+	"strings"
 )
 
 // DataPackage represents a compressed and optionally encrypted file that may
 // or may not exist on disk, yet.
+//
+// PackagePath is the full path to the data package file, which may or may not
+// exist, yet. If it is empty, the package will be Packed or Unpacked to
+// STDOUT or from STDIN, respectively.
+//
+// KeyPath is the full path to a file holding an ASCII-armored GPG key
+// for encryption or decryption (in which case the file must contain both
+// public and private keys).
+//
+// PublicKeyEmail is the email associated with a public key uploaded to a key
+// server and can be used in place of KeyPath for encryption. If both are
+// given, KeyPath is used instead.
+//
+// KeyPassPath is the full path to a file holding the password for the key.
+// The password can alternatively be exported to the PACKER_KEYPASS environment
+// variable. The environment variable is preferred if both are given.
```

Comment:
Ultimately, I think all this configuration complexity should be left up to the individual UI that makes use of the library. Which would mean that we should in fact be accepting Readers and ReadWriters instead of paths on the object. Do you think we should do that now?